server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -178,6 +178,10 @@ private synchronized void syncSegments(boolean isRetry) {
         if (indexShard.getReplicationTracker().isPrimaryMode() == false) {
             return;
         }
+        ReplicationCheckpoint checkpoint = indexShard.getLatestReplicationCheckpoint();
+        if (indexShard.indexSettings().isSegRepWithRemoteEnabled()) {
```

Comment:
Can remove the isSegRepWithRemoteEnabled check here? I don't think we are supporting node-node + remote store case.

Reply:
done.